EIPS/eip-5299.md

```diff
@@ -0,0 +1,168 @@
+---
+eip: 5299
+title: Layer3 scaling pattern
+description: A primitive to facilitate eternal upgradeability of smart contracts, by making storage slots dynamic.
+author: Vinay Kumar (@v9hstk), Pratik Agarwal (@Prtk418)
+discussions-to: https://ethereum-magicians.org/t/eip-9000-layer3-scaling-standard/9039
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-07-18
+requires: 2535
+---
+
+## Abstract
+Upgrading a storage layout is required in case of a complex project if it is re-architected. The EIP introduces dynamic storage slots, which can be upgraded without deploying a new contract.
+
+## Motivation
+With the existing smart contract upgradable design patterns available, while it’s possible to upgrade the code implementation, it is not possible to completely upgrade the storage structure, which requires a re-deployment of the contracts.
+
+From the end-user perspective, it is annoying to migrate from an old version of an app to a new version, especially if it requires deploying multiple transactions.
+
+
+## Specification
+This EIP is an extension of [EIP-2535](./eip-2535.md), adding a mechanism that maintains the old data points and enabling the possibility to update the structure of storage struct while updating the implementation.
+
+### Layer3 scaling standard
+For upgrading a storage structure we will store the slot addresses in a special struct we will call it ParentStorage and then retrieve the slot whenever the specific facet interacts with data store.
+
+Here is a simple example that shows dynamic slot based storage and its use in one of the implementation contract:
```

Comment:
I don't see an actual specification here.  The majority of this section appears to be an example implementation (which should go in the `## Reference Implementation` section) but no actual specification exists at the moment and that is required for an EIP.